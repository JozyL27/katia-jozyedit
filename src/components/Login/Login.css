.Login__container {
  width: 70%;
  font-size: 1.6rem;
  text-align: center;
  padding: 1rem 1rem 2rem;
}

.Login__header {
  margin: 1.5rem 0 2rem;
  font-size: 3rem;
}

.Login__submit-button-div {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 1.5rem;
}

.Login__signup-link {
  font-weight: bolder;
  color: rgb(48, 100, 153);
  cursor: pointer;
  font-size: 2.25rem;
  margin: 0 auto;
}

.Login__or {
  font-size: 2rem;
  margin: .75rem 0;
}

.demoContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 30px;
}

#Login__email, #Login__password {
  border-radius: 0;
  -webkit-border-radius:0px;
}