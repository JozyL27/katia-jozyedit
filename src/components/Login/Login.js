import React, { Component } from 'react'
import {Link} from 'react-router-dom'
import AuthApiService from '../../services/auth-api-service'
import './Login.css'

export default class Login extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = { error: null }

  handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    this.setState({error: null})
    const { email, password } = ev.target

    AuthApiService.postLogin({
      email: email.value.toLowerCase().trim(),
      password: password.value.trim()
    })
      .then(res => {
        email.value = ''
        password.value = ''
        this.props.onLoginSuccess(res.authToken)
      })
      .catch(res => {
        this.setState({error: res.error})
      })
  }

  render() {
    return (
      <>
      <main id='lp__main-flex'>
        <div className='lp__logoContainer'>
        <Link to='/' className='lp__lp-link'>Katia</Link>
        </div>
        <section className='Login__container'>
          <h2 className='Login__header'>Login</h2>
          <form
            className='Login__form'
            onSubmit={this.handleSubmitJwtAuth}
          >
            <div role='alert'>
              {this.state.error && <p className='Login__error'>{this.state.error}</p>}
            </div>
            <div className='Login__email-div lp__input-div'>
              <label className='lp__label' htmlFor='Login__email'>
                Email
              </label>
              <input
                required
                name='email'
                id='Login__email'
                className='lp__input'>
              </input>
            </div>
            <div className='Login__password-div lp__input-div'>
              <label className='lp__label' htmlFor='Login__password'>
                Password
              </label>
              <input
                required
                name='password'
                type='password'
                id='Login__password'
                className='lp__input'>
              </input>
            </div>
            <button className='blue-button blue-button-margin' type='submit'>
                Login
              </button>
            <div className='Login__submit-button-div'>
              <p className='Login__or'>Need an account?</p>
              <Link className='Login__signup-link' to='/signup'>Sign Up</Link>
            </div>
          </form>
        </section>
      </main>
      </>
    )
  }
}

