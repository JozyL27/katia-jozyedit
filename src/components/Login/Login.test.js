import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Login from './Login'

describe('<Login />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})