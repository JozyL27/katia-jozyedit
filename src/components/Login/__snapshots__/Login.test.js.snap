// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders the UI as expected 1`] = `
<main
  id="lp__main-flex"
>
  <div
    className="lp__logoContainer"
  >
    <a
      className="lp__lp-link"
      href="/"
      onClick={[Function]}
    >
      Katia
    </a>
  </div>
  <section
    className="Login__container"
  >
    <h2
      className="Login__header"
    >
      Login
    </h2>
    <form
      className="Login__form"
      onSubmit={[Function]}
    >
      <div
        role="alert"
      />
      <div
        className="Login__email-div lp__input-div"
      >
        <label
          className="lp__label"
          htmlFor="Login__email"
        >
          Email
        </label>
        <input
          className="lp__input"
          id="Login__email"
          name="email"
          required={true}
        />
      </div>
      <div
        className="Login__password-div lp__input-div"
      >
        <label
          className="lp__label"
          htmlFor="Login__password"
        >
          Password
        </label>
        <input
          className="lp__input"
          id="Login__password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="blue-button blue-button-margin"
        type="submit"
      >
        Login
      </button>
      <div
        className="Login__submit-button-div"
      >
        <p
          className="Login__or"
        >
          Need an account?
        </p>
        <a
          className="Login__signup-link"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </div>
    </form>
  </section>
</main>
`;
